.github/workflows/docs.yml

```
@@ -0,0 +1,47 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name: Docs
+
+on:
+  pull_request:
+    branches:
+      - master
+
+concurrency:
+  group: dep-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  sphinx:
+    name: sphinx-build
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+          cache: 'pip'
```

```suggestion
          cache: 'pip'
          cache-dependency-path: docs/requirements.txt
```
The caching relies on cache-dependency-path which is default to requirements.txt ( https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages)